core/src/main/java/hudson/triggers/Trigger.java

```diff
@@ -264,21 +264,25 @@ public void run(AbstractProject p) {
         // Process all triggers, except SCMTriggers when synchronousPolling is set
         for (ParameterizedJobMixIn.ParameterizedJob p : inst.getAllItems(ParameterizedJobMixIn.ParameterizedJob.class)) {
             for (Trigger t : p.getTriggers().values()) {
-                if (! (t instanceof SCMTrigger && scmd.synchronousPolling)) {
-                    LOGGER.log(Level.FINE, "cron checking {0} with spec ‘{1}’", new Object[] {p, t.spec.trim()});
-
-                    if (t.tabs.check(cal)) {
-                        LOGGER.log(Level.CONFIG, "cron triggered {0}", p);
-                        try {
-                            t.run();
-                        } catch (Throwable e) {
-                            // t.run() is a plugin, and some of them throw RuntimeException and other things.
-                            // don't let that cancel the polling activity. report and move on.
-                            LOGGER.log(Level.WARNING, t.getClass().getName() + ".run() failed for " + p, e);
+                if (t !=null && t.spec != null && t.tabs != null) {
+                    if (!(t instanceof SCMTrigger && scmd.synchronousPolling)) {
+                        LOGGER.log(Level.FINE, "cron checking {0} with spec ‘{1}’", new Object[]{p, t.spec.trim()});
+
+                        if (t.tabs.check(cal)) {
+                            LOGGER.log(Level.CONFIG, "cron triggered {0}", p);
+                            try {
+                                t.run();
+                            } catch (Throwable e) {
+                                // t.run() is a plugin, and some of them throw RuntimeException and other things.
+                                // don't let that cancel the polling activity. report and move on.
+                                LOGGER.log(Level.WARNING, t.getClass().getName() + ".run() failed for " + p, e);
+                            }
+                        } else {
+                            LOGGER.log(Level.FINER, "did not trigger {0}", p);
                         }
-                    } else {
-                        LOGGER.log(Level.FINER, "did not trigger {0}", p);
                     }
+                } else {
+                    LOGGER.log(Level.WARNING, "The job {0} has a syntactically incorrect config and is missing the cron spec for a trigger", p.getFullName());
```

Comment:
the spec is optional for a Trigger - it's only an issue where this is an SCMTrigger and we are using polling IIUC.
Hence the check should be inside `if (!(t instanceof SCMTrigger && scmd.synchronousPolling))` rather than outside it.